src/request.jl

```diff
@@ -41,6 +41,9 @@ end
   ui, ft, num_requested, wp = __request_threads(num_requested, wp, _first(threadmask))
   (ui, ), (ft, )
 end
+@inline function _request_threads(num_requested::UInt32, wp::Ptr, i::Int, threadmask) # fallback in absence of static scheduling
+    _request_threads(num_requested::UInt32, wp::Ptr, StaticInt{i}(), threadmask)
```

Comment:
This is obviously bad.
I suggest using `CPUSummary.sys_threads()`:
```julia
julia> CPUSummary.sys_threads()
static(251)

julia> cld(CPUSummary.sys_threads(), static(64))
static(4)
```
This method will only recompile when people change `Sys.CPU_THREADS`, which should not be happening often.
Then, you can do an unrolled branch-dispatch from `cld(CPUSummary.sys_threads(), static(64))` to `1` to pick the number.

This shouldn't be many branches, and branches are way faster than dynamic dispatches.

How do unroll a static number of branches?
Two ideas:
1. Recursion.
```julia
function gettup() # note that this is of length m - 1, because we drop the `1`
    m = cld(CPUSummary.sys_threads(), static(64))
    ntuple(static∘Base.Fix1(-,m+1), Val(Int(m)-1))
end
function bar(i)
    check_match(gettup(), i)
end
function check_match(x::Tuple{StaticInt{N},Vararg}, i::Int) where {N}
     i == N && return foo(static(N))
     check_match(Base.tail(x), i)
end
check_match(::Tuple{}, i::Int) = foo(static(1))
```
2. Use an `@generated` function.
```julia
function bar(i)
    check_match(cld(CPUSummary.sys_threads(), static(64)), i)
end
@generated function check_match(::StaticInt{N}, i) where {N}
    quote
        Base.Cartesian.@nif $N n -> $(N+1) - n == i n -> foo(static(n))
    end
end
```
I've been preferring recursion in the code I've written recently.

Reply:
But I think this recursion should be simple enough that the compiler can tell it terminates on its own.
Granted, I'd have said the same about all the cases inside SimpleChains...

Reply:
Obviously replace `foo` (`_request_threads`) and `bar` (`request_threads`) as appropriate.